### PR TITLE
[DI] Deprecate Container::initialized() for privates

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes-ish
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

See https://github.com/symfony/symfony/pull/22801#issuecomment-302875014

Failing test seems unrelated.

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Container.php [hunk 1]
```diff
@@ -403,7 +407,7 @@ public function getServiceIds()
         }
         $ids[] = 'service_container';
 
-        return array_unique(array_merge($ids, array_keys($this->methodMap), array_keys($this->services)));
+        return array_values(array_unique(array_merge($ids, array_keys($this->methodMap), array_keys($this->services))));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Container.php]: what is this for? related?

Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/Container.php]: i noted with https://github.com/symfony/symfony/pull/22801/files#diff-8aeba2b88d0347062017ee160a6beb49R141 keys are preserved; i want to do a clean assertion against `getServiceIds`, not relying on `array_values` or indexes in tests.

But can be left out yes; i just used this PR make the change as of 3.4 :)

Do you want a separate PR or rely on the current behavior?

Comment (ro0NL) [inline src/Symfony/Component/DependencyInjection/Container.php]: Reverted. not yet an issue in 3.4.

**Comment by nicolas-grekas** [comment]
Thank you @ro0NL.
